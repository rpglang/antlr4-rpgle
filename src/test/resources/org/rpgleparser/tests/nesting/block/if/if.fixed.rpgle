 
     C                   IF        isTest()
     C                   ENDIF
     C                   IF        %ERROR()
     C                   IF        %STATUS(CUST)=1211
     C                   OPEN      CUST
     C                   ELSEIF    %STATUS(CUST)=1221
     C                   READ(E)   CUST
     C                   EVAL      CSNAME='CCC'
     C                   UPDATE(E) CUSTR
     C                   ENDIF
     C                   ENDIF 