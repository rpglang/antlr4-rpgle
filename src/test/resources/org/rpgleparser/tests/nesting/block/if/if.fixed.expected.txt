/*===TOKENS===*/
CS_FIXED                 <C>
BlankIndicator           <  >
BlankFlag                < >
BlankIndicator           <  >
'              '         <              >
OP_IF                    <IF>
ID                       <isTest>
'('                      <(>
')'                      <)>
C_FREE_NEWLINE           <
>
CS_FIXED                 <C>
BlankIndicator           <  >
BlankFlag                < >
BlankIndicator           <  >
'              '         <              >
OP_ENDIF                 <ENDIF>
'              '         <              >
'              '         <              >
CS_FieldLength           <     >
CS_DecimalPositions      <  >
BlankIndicator           <  >
BlankIndicator           <  >
BlankIndicator           <  >
EOL                      <
>
CS_FIXED                 <C>
BlankIndicator           <  >
BlankFlag                < >
BlankIndicator           <  >
'              '         <              >
OP_IF                    <IF>
BIF_ERROR                <%ERROR>
'('                      <(>
')'                      <)>
C_FREE_NEWLINE           <
>
CS_FIXED                 <C>
BlankIndicator           <  >
BlankFlag                < >
BlankIndicator           <  >
'              '         <              >
OP_IF                    <IF>
BIF_STATUS               <%STATUS>
'('                      <(>
ID                       <CUST>
')'                      <)>
'='                      <=>
NUMBER                   <1211>
C_FREE_NEWLINE           <
>
CS_FIXED                 <C>
BlankIndicator           <  >
BlankFlag                < >
BlankIndicator           <  >
'              '         <              >
OP_OPEN                  <OPEN>
CS_FactorContent         <CUST>
'              '         <              >
CS_FieldLength           <     >
CS_DecimalPositions      <  >
BlankIndicator           <  >
BlankIndicator           <  >
BlankIndicator           <  >
EOL                      <
>
CS_FIXED                 <C>
BlankIndicator           <  >
BlankFlag                < >
BlankIndicator           <  >
'              '         <              >
OP_ELSEIF                <ELSEIF>
BIF_STATUS               <%STATUS>
'('                      <(>
ID                       <CUST>
')'                      <)>
'='                      <=>
NUMBER                   <1221>
C_FREE_NEWLINE           <
>
CS_FIXED                 <C>
BlankIndicator           <  >
BlankFlag                < >
BlankIndicator           <  >
'              '         <              >
OP_READ                  <READ>
'('                      <(>
CS_OperationAndExtender  <E>
')'                      <)>
CS_FactorContent         <CUST>
'              '         <              >
CS_FieldLength           <     >
CS_DecimalPositions      <  >
BlankIndicator           <  >
BlankIndicator           <  >
BlankIndicator           <  >
EOL                      <
>
CS_FIXED                 <C>
BlankIndicator           <  >
BlankFlag                < >
BlankIndicator           <  >
'              '         <              >
OP_EVAL                  <EVAL>
ID                       <CSNAME>
'='                      <=>
StringLiteralStart       <'>
StringContent            <CCC>
StringLiteralEnd         <'>
C_FREE_NEWLINE           <
>
CS_FIXED                 <C>
BlankIndicator           <  >
BlankFlag                < >
BlankIndicator           <  >
'              '         <              >
OP_UPDATE                <UPDATE>
'('                      <(>
CS_OperationAndExtender  <E>
')'                      <)>
CS_FactorContent         <CUSTR>
'              '         <              >
CS_FieldLength           <     >
CS_DecimalPositions      <  >
BlankIndicator           <  >
BlankIndicator           <  >
BlankIndicator           <  >
EOL                      <
>
CS_FIXED                 <C>
BlankIndicator           <  >
BlankFlag                < >
BlankIndicator           <  >
'              '         <              >
OP_ENDIF                 <ENDIF>
'              '         <              >
'              '         <              >
CS_FieldLength           <     >
CS_DecimalPositions      <  >
BlankIndicator           <  >
BlankIndicator           <  >
BlankIndicator           <  >
EOL                      <
>
CS_FIXED                 <C>
BlankIndicator           <  >
BlankFlag                < >
BlankIndicator           <  >
'              '         <              >
OP_ENDIF                 <ENDIF>
'              '         <              >
'              '         <              >
CS_FieldLength           <     >
CS_DecimalPositions      <  >
BlankIndicator           <  >
BlankIndicator           <  >
BlankIndicator           <  >
/*===TREE===*/
(r
  (statement
    (block
      (ifstatement
        (beginif
          C
          (cs_controlLevel '  ')
          (onOffIndicatorsFlag ' ')
          (cs_indicators '  ')
          (factor '              ')
          IF
          (c_free
            (expression
              (function (functionName (free_identifier (idOrKeyword isTest))) (args ( )))
            )
          )
          \r\n
        )
        (endif
          C
          (cs_controlLevel '  ')
          (onOffIndicatorsFlag ' ')
          (cs_indicators '  ')
          (factor '              ')
          (csENDIF
            ENDIF
            (cspec_fixed_standard_parts
              (factor '              ')
              (resultType '              ')
              '     '
              '  '
              (resultIndicator '  ')
              (resultIndicator '  ')
              (resultIndicator '  ')
              \r\n
            )
          )
        )
      )
    )
  )
  (statement
    (block
      (ifstatement
        (beginif
          C
          (cs_controlLevel '  ')
          (onOffIndicatorsFlag ' ')
          (cs_indicators '  ')
          (factor '              ')
          IF
          (c_free (expression (bif (bif_error %ERROR ( )))))
          \r\n
        )
        (statement
          (block
            (ifstatement
              (beginif
                C
                (cs_controlLevel '  ')
                (onOffIndicatorsFlag ' ')
                (cs_indicators '  ')
                (factor '              ')
                IF
                (c_free
                  (expression
                    (expression
                      (bif
                        (bif_status %STATUS ( (identifier (free_identifier (idOrKeyword CUST))) ))
                      )
                    )
                    =
                    (expression (number 1211))
                  )
                )
                \r\n
              )
              (statement
                (cspec_fixed
                  C
                  (cs_controlLevel '  ')
                  (onOffIndicatorsFlag ' ')
                  (cs_indicators '  ')
                  (factor '              ')
                  (cspec_fixed_standard
                    (csOPEN
                      OPEN
                      (cspec_fixed_standard_parts
                        (factor (factorContent CUST))
                        (resultType '              ')
                        '     '
                        '  '
                        (resultIndicator '  ')
                        (resultIndicator '  ')
                        (resultIndicator '  ')
                        \r\n
                      )
                    )
                  )
                )
              )
              (elseifstmt
                C
                (cs_controlLevel '  ')
                (onOffIndicatorsFlag ' ')
                (cs_indicators '  ')
                (factor '              ')
                ELSEIF
                (c_free
                  (expression
                    (expression
                      (bif
                        (bif_status %STATUS ( (identifier (free_identifier (idOrKeyword CUST))) ))
                      )
                    )
                    =
                    (expression (number 1221))
                  )
                )
                \r\n
              )
              (statement
                (cspec_fixed
                  C
                  (cs_controlLevel '  ')
                  (onOffIndicatorsFlag ' ')
                  (cs_indicators '  ')
                  (factor '              ')
                  (cspec_fixed_standard
                    (csREAD
                      READ
                      (cs_operationExtender ( E ))
                      (cspec_fixed_standard_parts
                        (factor (factorContent CUST))
                        (resultType '              ')
                        '     '
                        '  '
                        (resultIndicator '  ')
                        (resultIndicator '  ')
                        (resultIndicator '  ')
                        \r\n
                      )
                    )
                  )
                )
              )
              (statement
                (cspec_fixed
                  C
                  (cs_controlLevel '  ')
                  (onOffIndicatorsFlag ' ')
                  (cs_indicators '  ')
                  (factor '              ')
                  (cspec_fixed_standard
                    (csEVAL
                      EVAL
                      (c_free
                        (expression
                          (expression (identifier (free_identifier (idOrKeyword CSNAME))))
                          =
                          (expression (literal ' CCC '))
                        )
                      )
                      \r\n
                    )
                  )
                )
              )
              (statement
                (cspec_fixed
                  C
                  (cs_controlLevel '  ')
                  (onOffIndicatorsFlag ' ')
                  (cs_indicators '  ')
                  (factor '              ')
                  (cspec_fixed_standard
                    (csUPDATE
                      UPDATE
                      (cs_operationExtender ( E ))
                      (cspec_fixed_standard_parts
                        (factor (factorContent CUSTR))
                        (resultType '              ')
                        '     '
                        '  '
                        (resultIndicator '  ')
                        (resultIndicator '  ')
                        (resultIndicator '  ')
                        \r\n
                      )
                    )
                  )
                )
              )
              (endif
                C
                (cs_controlLevel '  ')
                (onOffIndicatorsFlag ' ')
                (cs_indicators '  ')
                (factor '              ')
                (csENDIF
                  ENDIF
                  (cspec_fixed_standard_parts
                    (factor '              ')
                    (resultType '              ')
                    '     '
                    '  '
                    (resultIndicator '  ')
                    (resultIndicator '  ')
                    (resultIndicator '  ')
                    \r\n
                  )
                )
              )
            )
          )
        )
        (endif
          C
          (cs_controlLevel '  ')
          (onOffIndicatorsFlag ' ')
          (cs_indicators '  ')
          (factor '              ')
          (csENDIF
            ENDIF
            (cspec_fixed_standard_parts
              (factor '              ')
              (resultType '              ')
              '     '
              '  '
              (resultIndicator '  ')
              (resultIndicator '  ')
              (resultIndicator '  ')
              <EOF>
            )
          )
        )
      )
    )
  )
)
/*======*/